src/main/scala/is/hail/methods/ImputeSexPlink.scala

```diff
@@ -1,90 +1,39 @@
 package is.hail.methods
 
-import is.hail.annotations._
 import is.hail.expr._
-import is.hail.stats.InbreedingCombiner
 import is.hail.utils._
-import is.hail.variant.{GenomeReference, Genotype, Locus, VariantDataset}
+import is.hail.variant.{GenomeReference, Locus, VariantSampleMatrix}
 
 object ImputeSexPlink {
-
-  def schema: Type = TStruct("isFemale" -> TBoolean(),
-    "Fstat" -> TFloat64(),
-    "nTotal" -> TInt64(),
-    "nCalled" -> TInt64(),
-    "expectedHoms" -> TFloat64(),
-    "observedHoms" -> TInt64())
-
-  def determineSex(ibc: InbreedingCombiner, fFemaleThreshold: Double, fMaleThreshold: Double): Option[Boolean] = {
-    ibc.Fstat
-      .flatMap { x =>
-        if (x < fFemaleThreshold)
-          Some(true)
-        else if (x > fMaleThreshold)
-          Some(false)
-        else None
-      }
-  }
-
-  def apply(vds: VariantDataset,
+  def apply(in: VariantSampleMatrix,
     mafThreshold: Double,
     includePar: Boolean,
     fMaleThreshold: Double,
     fFemaleThreshold: Double,
-    popFrequencyExpr: Option[String]): Map[Annotation, Annotation] = {
+    popFrequencyExpr: Option[String]): VariantSampleMatrix = {
+    var vsm = in
 
-    val query = popFrequencyExpr.map { code =>
-      val (t, f) = vds.queryVA(code)
-      t match {
-        case _: TFloat64 => f
-        case other => fatal(s"invalid population frequency.  Expected Double, but got `$other'")
-      }
-    }
-
-    val gr = vds.vSignature match {
+    val gr = vsm.vSignature match {
       case TVariant(x, _) => x.asInstanceOf[GenomeReference]
     }
 
     val xIntervals = IntervalTree(gr.xContigs.map(contig => Interval(Locus(contig, 0), Locus(contig, gr.contigLength(contig)))).toArray)
-    var xVds = vds.filterIntervals(xIntervals, keep = true)
+    vsm = vsm.filterIntervals(xIntervals, keep = true)
 
     if (!includePar)
-      xVds = xVds.filterIntervals(IntervalTree(gr.par), keep = false)
+      vsm = vsm.filterIntervals(IntervalTree(gr.par), keep = false)
+
+    vsm = vsm.annotateVariantsExpr(
+      s"va = ${ popFrequencyExpr.getOrElse("gs.map(g => g.nNonRefAlleles).sum() / gs.filter(g => g.isCalled).count() / 2") }")
+
+    val resultSA = vsm
```

Comment:
I'm very excited about the direction of this change.  The code got 80% smaller and infinitely clearer.  It will be even better in Python.  No more mucking around with `Option`s and queriers and whatnot.  I see tons of stuff that can move into Python now, esp. the tests.